### DISCUSSION TITLE
Does Typst support range condition for numbers (like Python)?

### DISCUSSION THREAD

**Discussion by Andrew15-5** [typst/typst#2183] [Q&A] [answered]
I didn't find information about double comparison like this: `-5 <= x < 10` in the documentation. Is it supported (v0.8)? AFAIK, Rust does have this feature (as well as Python), but most other languages does not.

A similar question is about a very rare but quite <strike>nice</strike>interesting use case: `0 < a < b < c < 69`. It would be so cool if this is supported.

Either way, the documentation should clarify this usage.

**Update**:
Created an issue: #2184.

**Comment by Andrew15-5** [comment, marked as answer]
Writing `-5 < 0 < 10` gives an error: `cannot compare boolean and integer (typst)`. So, the answer is **no, it doesn't**.

Can I open a feature request for this one? Or this will never be implemented? Is there a reason why it's not already implemented?

**Reply by Dherse** [reply]
While I am only a contributor, I do like the idea of having that operator since I generally find it easier to read than `-5 < x && x < 10`

**Comment by laurmaedje** [comment]
Rust says "comparison operators cannot be chained" if I do this. But I think it's a nice feature, so feel free to open a feature request.
